modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java

```diff
@@ -213,7 +213,7 @@ private BeanDefinition resolveComponent(ComponentModel parentComponentModel, Spr
             addAnnotation(ANNOTATION_NAME, componentModel.getIdentifier(), componentModel);
             // We need to use a mutable map since spring will resolve the properties placeholder present in the value if needed
             // and it will be done by mutating the same map.
-            addAnnotation(ANNOTATION_PARAMETERS, new HashMap<>(componentModel.getParameters()), componentModel);
+            addAnnotation(ANNOTATION_PARAMETERS, new HashMap<>(componentModel.getFilteredParameters()), componentModel);
```

Comment:
better to make a check by the identifier of the component and filtering the params here that spreading that logic to the ComponentModel.